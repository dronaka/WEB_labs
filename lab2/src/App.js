import React from "react";
import Cities from "./components/Cities";
import Geolocation from "./components/Geolocation"
import "./App.css"

class App extends React.Component {
  render() {
    return (
      <div>
        <Geolocation />
        <Cities />
      </div>
    );
  }
}


export default App;