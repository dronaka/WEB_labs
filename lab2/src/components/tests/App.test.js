import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

import App from "../../App";


describe("<App /> render", () => {

  const storeCreator = configureStore();
  const DEFAULT_STORE = storeCreator({
    cities: {
      cities: []
    },
    geo: {}
  });

  test("should has <Favorites /> and <Geolocation /> with empty state", () => {
    const tree = renderer
      .create(
        <Provider store={DEFAULT_STORE}>
          <App />
        </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});