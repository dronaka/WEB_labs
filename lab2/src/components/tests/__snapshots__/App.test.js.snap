// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render should has <Favorites /> and <Geolocation /> with empty state 1`] = `
<div>
  <div
    className="geolocation"
  >
    <h1
      className="header"
    >
      Refresh geolocation
    </h1>
    <button
      className="button"
      onClick={[Function]}
    >
      Get geolocation
    </button>
  </div>
  <div
    className="cities"
  >
    <h1>
      Ð¡ities
    </h1>
    <form
      class="add-city"
      onSubmit={[Function]}
    >
      <input
        class="input"
        name="cityName"
        placeholder="City name"
        required={true}
        type="text"
      />
      <input
        class="button"
        type="submit"
        value="Add to city"
      />
    </form>
    <div
      className="forecasts"
    />
  </div>
</div>
`;
